internal/decode/decode.go

```diff
@@ -0,0 +1,133 @@
+// Package decode implements a generic interface{} decoder. The intention is
+// to use it to decode arbitrary map[interface{}]interface{} objects into
+// structs or other complex objects.
+package decode
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+const _tagName = "config"
+
+var _typeOfDecoder = reflect.TypeOf((*Decoder)(nil)).Elem()
+
+// Into is a function that attempts to decode the source data into the given
+// destination. dst MUST be a pointer to a value.
+//
+// 	var (
+// 		decode decode.Into = ...
+// 		value map[string]MyStruct
+// 	)
+// 	err := decode(&value)
+type Into func(dst interface{}) error
```

Comment:
Suggest renaming `dst` to `dest` - losing too much info for the sake of brevity.